### PR TITLE
InstructorFeedbackResultsPageUiTest: improve stability of hover-triggered popover image verification #7686

### LINKED ISSUE #7686: InstructorFeedbackResultsPageUiTest: improve stability of hover-triggered popover image verification
The `testViewPhotoAndAjaxForLargeScaledSession` test in `InstructorFeedbackResultsPageUiTest` is still moderately unstable even after prior work done in #7068 and #7083. This is mainly because popovers that are triggered by hover actions sometimes fail to appear, causing hover-triggered popover image verifications to fail unpredictably.

Rather than retrying the entire test, `RetryManager` can be used to retry just the hover-verify process. This would help improve the overall success rate of the test.

### PR DESCRIPTION
**Pull request by whipermr5**
Fixes #7686

**Outline of Solution**
- Use `RetryManager` to retry hover-click process (fixes main `NoSuchElementException` and occasional `StaleElementReferenceException`)

**Other Stability Improvements**
- Use `waitForElementPresence` to get link element in `clickViewPhotoLink` (fixes occasional `TimeoutException`)
- Handle case when AJAX loader GIF URL is encountered instead of profile photo URL (fixes occasional `AssertionError`)

**Stability Measure**
- Number of CI builds where `InstructorFeedbackResultsPageUiTest`'s `testViewPhotoAndAjaxForLargeScaledSession` passed on first try: `55`/`55`

### CODE REVIEW

**Comment by whipermr5** [comment]
`RetryManager` is being moved from `test` to `main` in #7689; placing this on hold till that is merged.

File: src/test/java/teammates/test/pageobjects/InstructorFeedbackResultsPage.java [hunk 1]
```diff
@@ -372,10 +376,36 @@ public void hoverClickAndViewPhotoOnTableCell(int questionBodyIndex, int tableRo
...
+            final String popoverSelector, final String hoverSelector, final String urlRegex, String taskName) {
+        RetryManager.runUntilNoException(new RetryableTask(taskName) {
+            @Override
+            public void run() {
+                verifyPopoverImageUrl(popoverSelector, urlRegex);
+            }
+
+            @Override
+            public void beforeRetry() {
+                moveToElement(By.cssSelector(hoverSelector));
```
Comment (leeyimin) [inline src/test/java/teammates/test/pageobjects/InstructorFeedbackResultsPage.java]: If the current location is already at the element, will the moveToElement trigger the hover action? It seems like jquery hover action is only triggered on mouse enter and mouse leave, although I am not sure if it is the same action.

Comment (whipermr5) [inline src/test/java/teammates/test/pageobjects/InstructorFeedbackResultsPage.java]: Well, it seems to work (I've run it many times and on several runs, the retry does get triggered, and the first retry works).

**Comment by whipermr5** [comment]
After further testing, it seems the click-triggered popups also sometimes fail to appear and need retrying. @leeyimin requesting your review again for the latest commit 21082e9 😅

**Comment by whipermr5** [comment]
@damithc I added some code after your previous approval in 21082e9; seeking your reapproval :)

**Pull request merged**